### PR TITLE
Added method which return amount of vcpu by nodegroup and added the u…

### PR DESCRIPTION
**Pull request by betobetodev**
…sage of optimizer in Spark-Sahara plugin.

### CODE REVIEW

File: application_manager/plugins/sahara/plugin.py [hunk 1]
```diff
@@ -132,6 +136,23 @@ def start_application(self, data, spark_applications_ids, app_id):
...
+                                                               sahara,
+                                                               slave_ng)
+
+            cores, vms = optimizer.get_info(api.optimizer_url,
+                                            expected_time,
+                                            app_name,
+                                            days)
+
+            if cores > 0:
+                req_cluster_size = ((cores / cores_per_slave) +
```
Comment (igornatanael) [inline application_manager/plugins/sahara/plugin.py]: use math.ceil()

**Comment by betobetodev** [comment]
When application name is **bulma**, the field **days** should exists in submission of client. If name isn't **bulma**, check if exists **cluster size** and use it. 
In some (two) cases the Spark-Sahara will throw an exception: 
-> application name is **bulma** and the field **days** doesn't exists. 
-> the amount of **cores** is _less equal than zero._ and doesn't exists **cluster_size** parameter.

**Review by armstrongmsg — APPROVED** [review]

File: application_manager/plugins/sahara/plugin.py [hunk 2]
```diff
@@ -123,6 +122,15 @@ def start_application(self, data, spark_applications_ids, app_id):
...
             starting_cap = data['starting_cap']
 
+            # Optimizer Parameters
+            app_name = data['app_name']
+
+            if app_name.lower() == 'bulma':
+                if 'days' in data.keys():
+                    days = data['days']
+                else:
+                    raise ex.ConfigurationError()
```
Comment (igornatanael) [inline application_manager/plugins/sahara/plugin.py]: Please add a message for this exception

File: application_manager/plugins/sahara/plugin.py [hunk 3]
```diff
@@ -132,6 +140,27 @@ def start_application(self, data, spark_applications_ids, app_id):
...
+            cores, vms = optimizer.get_info(api.optimizer_url,
+                                            expected_time,
+                                            app_name,
+                                            days)
+
+            if cores <= 0:
+                if 'cluster_size' in data.keys():
+                    req_cluster_size = data['cluster_size']
+                else:
+                    raise ex.ConfigurationError()
```
Comment (igornatanael) [inline application_manager/plugins/sahara/plugin.py]: same here

**Review by igornatanael — CHANGES_REQUESTED** [review]

File: application_manager/plugins/sahara/plugin.py (hunk 2 above)
Comment (betobetodev) [reply application_manager/plugins/sahara/plugin.py]: Done @igornatanael .

File: application_manager/plugins/sahara/plugin.py (hunk 1 above)
Comment (betobetodev) [reply application_manager/plugins/sahara/plugin.py]: Done @igornatanael .

File: application_manager/plugins/sahara/plugin.py (hunk 3 above)
Comment (betobetodev) [reply application_manager/plugins/sahara/plugin.py]: Done @igornatanael .

**Review by igornatanael — APPROVED** [review]

**Pull request merged**